File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/version/VersionViewModelFactory.java
package org.dvbviewer.controller.data.version;

import android.app.Application;
import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;

public class VersionViewModelFactory extends ViewModelProvider.NewInstanceFactory {

    private Application mApplication;
    private VersionRepository mRepo;

    public VersionViewModelFactory(Application application, VersionRepository repo) {
        mApplication = application;
        mRepo = repo;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new VersionViewModel(mApplication, mRepo);
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/version/VersionViewModelFactory.kt
package org.dvbviewer.controller.data.version

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class VersionViewModelFactory(private val mApplication: Application, private val mRepo: VersionRepository) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return VersionViewModel(mApplication, mRepo) as T
    }

}